Lab 8.ipynb

```diff
@@ -0,0 +1,652 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "fig, ax = plt.subplots(3, 1, figsize = (8,15))\n",
+    "plt.close()\n",
+    "fig2, ax2 = plt.subplots(1, 1)\n",
+    "\n",
+    "n_higgs = (100 / (100 + 20000)) * (len(high_lumi['mass']) / len(higgs['mass']))\n",
+    "n_qcd = (20000 / (100 + 20000)) * (len(high_lumi['mass']) / len(qcd['mass']))\n",
+    "\n",
+    "qcd_weight = np.array([n_qcd] * len(qcd['mass']))\n",
+    "higgs_weight = np.array([n_higgs] * len(higgs['mass']))\n",
+    "\n",
+    "mass_lower = qcd['mass'].min()\n",
+    "mass_upper = qcd['mass'].max()\n",
+    "\n",
+    "lower_val = qcd['d2'].min()\n",
+    "upper_val = qcd['d2'].max()\n",
+    "\n",
+    "bin_num = 100\n",
+    "\n",
+    "hist_qcd = ax[0].hist2d(qcd['mass'], qcd['d2'], bins = [95, bin_num], range = [[mass_lower, mass_upper], [lower_val, upper_val]], density = False, weights = qcd_weight)\n",
+    "hist_higgs = ax[1].hist2d(higgs['mass'], higgs['d2'], bins = [95, bin_num], range = [[mass_lower, mass_upper], [lower_val, upper_val]], density = False, weights = higgs_weight)\n",
+    "hist_lumi = ax[2].hist2d(high_lumi['mass'], high_lumi['d2'], bins = [95, bin_num], range = [[mass_lower, mass_upper], [lower_val, upper_val]], density = False)\n",
+    "\n",
+    "qcd_slice = hist_qcd[0][35]\n",
+    "higgs_slice = hist_higgs[0][35]\n",
+    "lumi_slice = hist_lumi[0][35]\n",
+    "\n",
+    "plt.fill_between(hist_qcd[2][0:30], qcd_slice[0:30] + higgs_slice[0:30], color = 'blue', alpha = 1, step = 'pre')\n",
+    "plt.fill_between(hist_lumi[2][0:30], lumi_slice[0:30], color = 'pink', alpha = .7, step = 'pre')\n",
+    "\n",
+    "ax2.step(hist_qcd[2][0:30], qcd_slice[0:30] + higgs_slice[0:30], alpha = 1, color = 'blue')\n",
+    "ax2.step(hist_lumi[2][0:30], lumi_slice[0:30], color = 'pink', alpha = .9)\n",
+    "ax2.set_title(\"Slice of d2 with Mass Cuts\")\n",
+    "ax2.legend([\"QCD and Higgs Stacked\", 'High Luminosity'])\n",
+    "\n",
+    "n_exp = lumi_slice[0]\n",
+    "n_sim = qcd_slice[0]\n",
+    "sig = stats.norm.ppf(stats.poisson.cdf(n_sim, n_exp, loc = 0))\n",
```

Comment:
Pretty sure your n_sim and n_exp should be reversed here, it's stats.poisson.cdf(value, mean)